lib/wallet/http.js

```diff
@@ -833,6 +783,405 @@ class HTTP extends Server {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      const passphrase = valid.str('passphrase');
+      const broadcast = valid.bool('broadcast', true);
+      const sign = valid.bool('sign', true);
+
+      assert(broadcast ? sign : true, 'Must sign when broadcasting.');
+
+      if (!name) {
+        const tx = await req.wallet.sendRevealAll();
+        return tx.getJSON(this.network);
+      }
+
+      const options = TransactionOptions.fromValidator(valid);
+      const mtx = await req.wallet.createReveal(name, options);
+
+      if (broadcast) {
+        const tx = await req.wallet.sendMTX(mtx, passphrase);
+        return res.json(200, tx.getJSON(this.network));
+      }
+
+      if (sign)
+        await req.wallet.sign(mtx, passphrase);
+
+      return res.json(200, mtx.getJSON(this.network));
+    });
+
+    // Create Redeem
+    this.post('/wallet/:id/redeem', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      const passphrase = valid.str('passphrase');
+      const broadcast = valid.bool('broadcast', true);
+      const sign = valid.bool('sign', true);
+
+      assert(broadcast ? sign : true, 'Must sign when broadcasting.');
+
+      if (!name) {
+        const tx = await req.wallet.sendRedeemAll();
+        return tx.getJSON(this.network);
+      }
+
+      const options = TransactionOptions.fromValidator(valid);
+      const mtx = await req.wallet.createRedeem(name, options);
+
+      if (broadcast) {
+        const tx = await req.wallet.sendMTX(mtx, passphrase);
+        return res.json(200, tx.getJSON(this.network));
+      }
+
+      if (sign)
+        await req.wallet.sign(mtx, passphrase);
+
+      return res.json(200, mtx.getJSON(this.network));
+    });
+
+    // Create Update
+    this.post('/wallet/:id/update', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      const data = valid.obj('data');
+      const passphrase = valid.str('passphrase');
+      const broadcast = valid.bool('broadcast', true);
+      const sign = valid.bool('sign', true);
+
+      assert(broadcast ? sign : true, 'Must sign when broadcasting.');
+      assert(name, 'Must pass name.');
+      assert(data, 'Must pass data.');
+
+      let resource;
+      try {
+        resource = Resource.fromJSON(data);
+      } catch (e) {
+        return res.json(400);
+      }
+
+      const options = TransactionOptions.fromValidator(valid);
+      const mtx = await req.wallet.createUpdate(name, resource, options);
+
+      if (broadcast) {
+        const tx = await req.wallet.sendMTX(mtx, passphrase);
+        return res.json(200, tx.getJSON(this.network));
+      }
+
+      if (sign)
+        await req.wallet.sign(mtx, passphrase);
+
+      return res.json(200, mtx.getJSON(this.network));
+    });
+
+    // Create Renewal
+    this.post('/wallet/:id/renewal', async (req, res) => {
```

Comment:
`renewal` vs `renew`?

Reply:
I'm pro -> use nouns in the URL and verbs as the http request type.

I think ```POST wallet renewal```, flows better than ```POST wallet renew``` since renewal is a noun. That being said, I'm not sure how the URLs in the node http calls are designed so I'd have to double check if that's a common thing to do.